Type alias cannot be a reference

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** 0.2.2
**OS:** Linux

In this example I have two types that are equal:
1. `&char`
1. `ALcharptr`

When translated into C, both types (although equal) will be passed to `char_vstring` differently:
1. `char_vstring(s)`
1. `char_vstring(&s)`

Full exampled [here](https://github.com/vlang/v/files/6292006/alias_bug.zip)(zip).

### V snippet
```v
type ALcharptr = &char
fn C.return_text_one() ALcharptr
fn C.return_text_two() &char

fn main() {
	println('type &char\talias ALcharptr\t:\t${get_one()}')
	println('type &char\talias none\t:\t${get_two()}')
}

fn get_one() string {
	s := C.return_text_one()
	return unsafe { cstring_to_vstring(s) }
}

fn get_two() string {
	s := C.return_text_two()
	return unsafe { cstring_to_vstring(s) }
}
```

### C snippet
```c
void main__main(void) {
	println(_STR("type &char\talias ALcharptr\t:\t%.*s", 1, main__get_one()));
	println(_STR("type &char\talias none\t:\t%.*s", 1, main__get_two()));
}

string main__get_one(void) {
	main__ALcharptr s = return_text_one();
	 string _t101 = char_vstring(&s); // <- Nein!
	return _t101;
}

string main__get_two(void) {
	char* s = return_text_two();
	 string _t102 = char_vstring(s);
	return _t102;
}
```

### Output
```
type &char      alias ALcharptr :       �lf
type &char      alias none      :       text
```

Cgen error on latest v:
```
PS D:\Games\Proekti\V\interactions> type test2.v  
type ALcharptr = &char

fn C.return_text_one() ALcharptr
fn C.return_text_two() &char

fn main() {
        println('type &char\talias ALcharptr\t:\t${get_one()}')
        println('type &char\talias none\t:\t${get_two()}')
}

fn get_one() string {
        s := C.return_text_one()
        return unsafe { cstring_to_vstring(s) }
}

fn get_two() string {
        s := C.return_text_two()
        return unsafe { cstring_to_vstring(s) }
}
PS D:\Games\Proekti\V\interactions> v version
V 0.4.4 2a68e2b
PS D:\Games\Proekti\V\interactions> v run test2.v 
==================
C:/Users/mclr/AppData/Local/Temp/v_0/test2.01HN390CQQ1AJEKT8Z3C0PW484.tmp.c:13009: warning: cast between pointer and integer of different size
tcc: error: undefined symbol 'return_text_one'
tcc: error: undefined symbol 'return_text_two'
...
==================
(Use `v -cg` to print the entire error message)

builder error:
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -2722,7 +2722,8 @@ fn (mut g Gen) ref_or_deref_arg(arg ast.CallArg, expected_type ast.Type, lang as
 						return
 					} else if arg.expr is ast.Ident && arg.expr.language == .c {
 						g.write('(voidptr)')
-					} else {
+					} else if !(!arg.is_mut && arg_sym.kind == .alias
+						&& g.table.unaliased_type(arg_typ).is_any_kind_of_pointer()) {
 						g.write('(voidptr)&/*qq*/')
 					}
 				} else {
@@ -2731,7 +2732,7 @@ fn (mut g Gen) ref_or_deref_arg(arg ast.CallArg, expected_type ast.Type, lang as
 						atype = g.unwrap_generic(atype)
 					}
 					if atype.has_flag(.generic) || arg.expr is ast.StructInit {
-						g.write('(voidptr)&/*qq*/')
+						g.write('(voidptr)&/*qq2*/')
 					} else if arg.expr is ast.None {
 						g.expr_with_opt(arg.expr, arg_typ, expected_type)
 						return

--- a/vlib/v/gen/c/testdata/alias_char_ptr_to_ptr.c.must_have
+++ b/vlib/v/gen/c/testdata/alias_char_ptr_to_ptr.c.must_have
@@ -1,0 +1,10 @@
+VV_LOCAL_SYMBOL string main__get_two(void) {
+char* s = main__return_text_two();
+string _t1 = cstring_to_vstring(s);
+return _t1;
+}
+VV_LOCAL_SYMBOL string main__get_one(void) {
+main__ALcharptr s = main__return_text_one();
+string _t1 = cstring_to_vstring(s);
+return _t1;
+}

--- a/vlib/v/gen/c/testdata/alias_char_ptr_to_ptr.vv
+++ b/vlib/v/gen/c/testdata/alias_char_ptr_to_ptr.vv
@@ -1,0 +1,17 @@
+type ALcharptr = &char
+fn return_text_one() ALcharptr {
+	return ALcharptr(''.str)
+}
+fn return_text_two() &char {
+	return ''.str
+}
+
+fn get_one() string {
+	s := return_text_one()
+	return unsafe { cstring_to_vstring(s) }
+}
+
+fn get_two() string {
+	s := return_text_two()
+	return unsafe { cstring_to_vstring(s) }
+}

